### ISSUE TITLE
Issue 1: Create Custom Optimizer for Qiskit-IonQ

### ISSUE THREAD

**Issue by Vadikus** [ionq-samples/Ion-Q-Thruster#1]
**Description**

Implement a custom optimizer for Qiskit specifically designed to work with IonQ's trapped-ion quantum computers. The optimizer should be able to transpile quantum circuits into the native gate set of IonQ's backend.

**Tasks**

- Initialize the IonQ provider and backend using Qiskit.
- Create a new optimization pass manager class that integrates with Qiskit's transpiler.
- Ensure that the optimizer can handle basic quantum circuits and convert them to the native IonQ gate set.

**References**

- [Qiskit Transpiler Pass Manager](https://docs.quantum.ibm.com/transpile)
- [IonQ Provider for Qiskit](https://github.com/qiskit-community/qiskit-ionq/blob/main/qiskit_ionq/ionq_provider.py)

**Comment by huyenemma** [comment]
Hello. I have a solution to this issue. Not yet implement the ion-trapped benefits but already can reduce circuit size by matrix optimization. [my solution](https://github.com/huyenemma/ionq_optimizer)

**Comment by splch** [comment]
> Hello. I have a solution to this issue. Not yet implement the ion-trapped benefits but already can reduce circuit size by matrix optimization. [my solution](https://github.com/huyenemma/ionq_optimizer)

thats so cool! ill add some more issues too for the challenges :)

**Comment by huyenemma** [comment]
Shall I make a pull request for issue 1?

**Comment by splch** [comment]
> Shall I make a pull request for issue 1?

yeah, though it might be more scalable to use a class like the qiskit transpiler docs describe to tackle the next 2 issues more easily!
